card/mod/01_core/set/all/rules.rb

```diff
@@ -32,6 +48,15 @@ def rule_card setting_code, options={}
 
 def rule_card_id setting_code, options={}
   fallback = options.delete( :fallback )
+  
+  if Card::Setting.user_specific? setting_code 
+    user_id = options[:user_id] || (options[:user] and options[:user].id) || Auth.current_id
+    if user_id
+      fallback = "#{setting_code}+#{Card[:all].id}"
+      setting_code = "#{setting_code}+#{user_id}"
+    end
+  end
+  
```

Comment:
this is cool. I'm tempted to find ways to break the user stuff with a parallel api around the word "preference", but I really like what you've done here.